action/rename.php

```diff
@@ -112,6 +113,48 @@ public function handle_ajax(Doku_Event $event) {
         }
     }
 
+    /**
+     * Handle media renames in media manager
+     *
+     * @param Doku_Event $event
+     * @return void
+     */
+    public function handleAjaxMediaManager(Doku_Event $event)
+    {
+        if ($event->data !== 'plugin_move_rename_mediamanager') return;
+
+        $event->preventDefault();
+        $event->stopPropagation();
+
+        global $INPUT;
+
+        $src = cleanID($INPUT->str('src'));
+        $dst = cleanID($INPUT->str('dst'));
+
+        /** @var helper_plugin_move_op $MoveOperator */
+        $MoveOperator = plugin_load('helper', 'move_op');
+
+        if ($src && $dst) {
```

Comment:
I'm missing a check either here or if we also don't want to allow this for admins, then in `helper_plugin_move_op`, if the file extension actually hasn't been changed. Not changing the file extension is relevant for security I think (to avoid changing a harmless file extension into a dangerous one) so this check needs to be done on the server side, it is not sufficient to disallow changing the file extension on the client side.